go-controller/pkg/ovn/obj_retry.go

```diff
@@ -1017,22 +1001,26 @@ func (oc *Controller) processObjectInTerminalState(objectsToRetry *retryObjs, ob
 // periodically or when explicitly requested.
 // Note: when applying WatchResource to a new resource type, the appropriate resource-specific logic must be added to the
 // the different methods it calls.
-func (oc *Controller) WatchResource(objectsToRetry *retryObjs) *factory.Handler {
+func (oc *Controller) WatchResource(objectsToRetry *retryObjs) (*factory.Handler, error) {
 	addHandlerFunc, err := oc.watchFactory.GetResourceHandlerFunc(objectsToRetry.oType)
 	if err != nil {
 		klog.Errorf("No resource handler function found for resource %v. "+
 			"Cannot watch this resource.", objectsToRetry.oType)
-		return nil
+		// TODO(FF): make this function able to return failures
```

Comment:
why cant this return an error in this commit?

Reply:
yeah, we can. I will change that!